### ISSUE TITLE
Could you make the built-in allegro select digital Midi and a default path/name for the patches?

### ISSUE THREAD

**Issue by i30817** [adventuregamestudio/ags#760]
The users would still need to download them but it would be much easier to explain.

**Comment by i30817** [comment]
Also, the message "Failed to detect MIDI driver DIGI; Error: 'DIGMID patch set not found'." is quite unhelpful because it doesn't mention what name it expects for the file and where it's looking.

**Comment by i30817** [comment]
On second check, my problem is another. Simply said, alsa midi through pulseaudio doesn't work with allegro (at least this version), and i was trying to go fully digital to work around it (which probably won't work because pulse controls everything audio, but it's worth a try).

I've seen the recent PR for https://github.com/adventuregamestudio/ags/pull/737 but the list of allegro drivers is not very clear (and the readme for the acsetup entries for these options wasn't updated with them).

So consider this issue invalid, unless you can do something about 'Also, the message "Failed to detect MIDI driver DIGI; Error: 'DIGMID patch set not found'." to be more clear.

**Comment by i30817** [comment]
Wait a second. I printed the output of the Allegro builtin (from cmake) on failure;

```
#include <stdio.h>

/* _digmid_find_patches:
 *  Tries to locate the GUS patch set directory and index file (default.cfg).
 */
int _digmid_find_patches(char *dir, int dir_size, char *file, int file_size)
{
   char filename[1024], tmp1[64], tmp2[64], tmp3[64], tmp4[64];
   AL_CONST char *name;
   char *datafile, *objectname, *envvar, *subdir, *s;

   name = get_config_string(uconvert_ascii("sound", tmp1), uconvert_ascii("patches", tmp2), NULL);
   datafile = uconvert_ascii("patches.dat", tmp1);
   objectname = uconvert_ascii("default.cfg", tmp2);
   envvar = uconvert_ascii("ULTRADIR", tmp3);
   subdir = uconvert_ascii("midi", tmp4);

   if (find_allegro_resource(filename, name, NULL, datafile, objectname, envvar, subdir, sizeof(filename)) != 0) {
   printf("---------------------------------\n");
    printf(filename);printf("\n");
    printf(name);printf("\n");
    printf(objectname);printf("\n");
    printf(envvar);printf("\n");
    printf(filename);printf("\n");
    printf(subdir);printf("\n");
    

    printf("---------------------------------\n");

      return FALSE;

   }

   if (dir && file) {
      s = ustrrchr(filename, '#');
      if(s == NULL)
         s = get_filename(filename);
      else
         s += ustrlen("#");
      ustrzcpy(file, file_size, s);

      usetc(s, 0);
      ustrzcpy(dir, dir_size, filename);
   }

   return TRUE;
}
```

It printed :
```

---------------------------------
/media/i30817/Huggin/Documents/projects/ags3/build/midi/default.cfg

default.cfg
ULTRADIR
/media/i30817/Huggin/Documents/projects/ags3/build/midi/default.cfg
midi
---------------------------------
Failed to detect MIDI driver DIGI; Error: 'DIGMID patch set not found'.
```

So what's happening here:

1. program tries to read the patches var from the loaded cfg file if any (not any here)
2. static build of allegro considers `ags` its executable and where it's at its 'home', and because there is no `patches=dir` available tries to read a sound font from `ags build dir/midi/default.cfg`? (not sure if this is the soundfont already or another cfg). Fails anyway.

**Comment by i30817** [comment]
Even weirder, if i place the patches.dat on the same dir as the ags executable, a different code path is taken that doesn't print. Instead it reports:
```

Sound settings: digital driver ID: 'ALSA' (0x414c5341), MIDI driver ID: 'DIGI' (0x44494749)
Unknown MIDI driver: DIGI, will try to find suitable one.

```

waaaat?

**Comment by ivan-mogilko** [comment]
> Even weirder, if i place the patches.dat on the same dir as the ags executable, a different code path is taken that doesn't print. Instead it reports:

This could be a mistake in ags itself after some recent changes, and I think I know what may be wrong.

**Comment by ivan-mogilko** [comment]
This seem to have fixed digmid detection: 83765735e774b31a751d35b1cd11e6e4366e0265
But with digmid driver the digital sounds no longer working. Not sure if this is how it supposed to be. I know that digmid will reuse "voices" from digital driver, but I expected it to leave at least enough for it to keep playing.

**Comment by i30817** [comment]
Unfortunate. And the path? Can it be on the same dir as the game, or does it have to be on the same path as the ags executable (which on linux can be in many places, often not writable).

**Comment by i30817** [comment]
What i'm doing right now is to setup all my games to put in

```
digiid=AUTO
midiid=NONE
```
in the acsetup.cfg. The few games that need midi are surprisingly recognizable from the silent main menu.

**Comment by i30817** [comment]
To be clear i think the root situation here is that the original games never expected the patches.dat file not to be placed on the same dir as the ags executable. In order to solve it, Allegro needs to get the game data dir as opposed to the executable dir.

**Comment by ivan-mogilko** [comment]
Closed in favor of #896.
